third_party/txt/src/txt/paragraph.cc

```diff
@@ -861,7 +861,11 @@ void Paragraph::Paint(SkCanvas* canvas, double x, double y) {
   canvas->translate(x, y);
   SkPaint paint;
   for (const PaintRecord& record : records_) {
-    paint.setColor(record.style().color);
+    if (record.style().has_foreground) {
+      paint = record.style().foreground;
+    } else {
+      paint.setColor(record.style().color);
```

Comment:
Call paint.reset() before paint.setColor() to clear any attributes besides the color that might have been set by a previous style with a foreground paint